cached_network_image/lib/src/image_provider/_image_loader.dart

```diff
@@ -98,43 +98,50 @@ class ImageLoader implements platform.ImageLoader {
           'CacheManager needs to be an ImageCacheManager. maxWidth and '
           'maxHeight will be ignored when a normal CacheManager is used.');
 
-      var stream = cacheManager is ImageCacheManager
-          ? cacheManager.getImageFile(url,
+      final stream = cacheManager is ImageCacheManager
+          ? cacheManager.getImageFile(
+              url,
               maxHeight: maxHeight,
               maxWidth: maxWidth,
               withProgress: true,
               headers: headers,
-              key: cacheKey)
-          : cacheManager.getFileStream(url,
-              withProgress: true, headers: headers, key: cacheKey);
+              key: cacheKey,
+            )
+          : cacheManager.getFileStream(
+              url,
+              withProgress: true,
+              headers: headers,
+              key: cacheKey,
+            );
 
-      await for (var result in stream) {
+      await for (final result in stream) {
         if (result is DownloadProgress) {
-          chunkEvents.add(ImageChunkEvent(
-            cumulativeBytesLoaded: result.downloaded,
-            expectedTotalBytes: result.totalSize,
-          ));
+          chunkEvents.add(
+            ImageChunkEvent(
+              cumulativeBytesLoaded: result.downloaded,
+              expectedTotalBytes: result.totalSize,
+            ),
+          );
         }
         if (result is FileInfo) {
-          var file = result.file;
-          var bytes = await file.readAsBytes();
-          var decoded = await decode(bytes);
+          final file = result.file;
+          final bytes = await file.readAsBytes();
+          final decoded = await decode(bytes);
           yield decoded;
         }
       }
-    } catch (e) {
+    } on Object catch (e) {
       // Depending on where the exception was thrown, the image cache may not
       // have had a chance to track the key in the cache at all.
       // Schedule a microtask to give the cache a chance to add the key.
       scheduleMicrotask(() {
         evictImage();
       });
 
-      errorListener?.call();
+      errorListener?.call(e);
       rethrow;
-    } finally {
-      await chunkEvents.close();
     }
+    await chunkEvents.close();
```

Comment:
linter swears at this, apparently he does not see the closure inside finally.

Reply:
Are you sure? I just added `finally` locally in your branch, but the linter is not complaining for me..

Reply:
Yes, I have same warning as [flutter_cache_manager](https://pub.dev/packages/flutter_cache_manager/score)

Reply:
In cache_manager that actually is an interesting one. I think it should be close in case of `subject != null && ignoreMemCache`

https://github.com/Baseflow/flutter_cache_manager/blob/develop/flutter_cache_manager/lib/src/web/web_helper.dart#L39-L41

However, in this case I think it's really a bug if it's not in finally.

Reply:
And what to do with it in the end?

Reply:
I think we really need to go back to having finally, so the stream is always closed.
```dart
    } finally {
      await chunkEvents.close();
    }
```

Reply:
ok